tests/test_energy_overflows.py

```diff
@@ -227,28 +227,27 @@ def compute_potential_energy(pot):
     test_energy = compute_potential_energy(potential)
     np.testing.assert_allclose(reference_energy, test_energy, rtol=rtol, atol=atol)
 
-    # If we call the two potentials separately, we will now see that both return inf energies
+    # If we call the two potentials separately, we will now see that both return nan energies
     all_pairs_energy = compute_potential_energy(all_pairs)
-    assert not np.isfinite(all_pairs_energy)
+    assert np.isnan(all_pairs_energy)
 
     pair_list_energy = compute_potential_energy(pair_list)
-    assert not np.isfinite(pair_list_energy)
+    assert np.isnan(pair_list_energy)
 
     # Make sure that the overflows are actually cancelling out
     pair_list_bound = pair_list.to_gpu(precision).bind(params).bound_impl
     all_pairs_bound = all_pairs.to_gpu(precision).bind(params).bound_impl
 
-    fixed_all_pairs_energy, all_pairs_overflows = all_pairs_bound.execute_fixed(x0, box)
-    fixed_pair_list_energy, pair_list_overflows = pair_list_bound.execute_fixed(x0, box)
-    assert all_pairs_overflows != 0
-    assert pair_list_overflows != 0
-    assert all_pairs_overflows + pair_list_overflows == 0
+    fixed_all_pairs_energy = all_pairs_bound.execute_fixed(x0, box)
+    fixed_pair_list_energy = pair_list_bound.execute_fixed(x0, box)
+    assert fixed_overflowed(fixed_all_pairs_energy)
+    assert fixed_overflowed(fixed_pair_list_energy)
 
-    # The sum of the fixed point energies should be bitwise identical to the test energy
-    assert FIXED_TO_FLOAT(fixed_all_pairs_energy + fixed_pair_list_energy) == test_energy
+    # The values are identical because of overflow, as its not clear how to go from
+    # C++ __int128 to python integer.
+    assert fixed_pair_list_energy == fixed_pair_list_energy
```

Comment:
Isn't this just testing to see if `fixed_pair_list_energy` is a `nan`? maybe just replace with `isnan(fixed_pair_list_energy)`

Reply:
This is wrong because it is comparing against itself, but it supposed to check that `fixed_pair_list_energy == fixed_all_pair_energy` because this is in fixed point and should be both represented by a LLONG_MAX

Reply:
Fixed the assertion here b6e7dc54317b3b0e60ad13417876516c8af5aae5